server/mqtt.go

```diff
@@ -242,6 +242,7 @@ type mqttAccountSessionManager struct {
 	flapTimer  *time.Timer                    // Timer to perform some cleanup of the flappers map
 	sl         *Sublist                       // sublist allowing to find retained messages for given subscription
 	retmsgs    map[string]*mqttRetainedMsgRef // retained messages
+	rmsCache   sync.Map                       // map[string(subject)]mqttRetainedMsg
```

Comment:
Should we not store pointers (`*mqttRetainedMsg`)?

Reply:
0/5 I chose to store a copy to simplify dealing with concurrent access. What would be the advantage of storing a pointer to the cloned object, as opposed to the object itself if either way they are wrapped as an interface{}?

Reply:
Yes, ok.